### ISSUE TITLE
Remove Putty from installer

### ISSUE THREAD

**Issue by hansfn** [gitextensions/gitextensions#5121]
I didn't find an issue for this, but it might have been discussed in other places.

GitExtensions recently removed Git and KDiff from the installer. ("The user will have to install external apps and tools separately.") I suggest that you do the same for Putty because:

1. You avoid the work of keeping Putty up-to-date in this project. 
2. Many will already have Putty installed.
3. If you keep your Putty up-to-date, you also have to update the one inside GitExtension (or change the settings so your version is used).

I could imagine that Putty was installed only if it wasn't already present, but Putty isn't needed to make key authentication work.

**Comment by vbjay** [comment]
I think a downloader that would either download the current installer or
the current portable would still be a god idea.  You don't know how many
use putty vs openssl.  I use putty all the time.  I pointed gitextensions
to use my installed putty instead of its putty.

**Comment by RussKie** [comment]
I don't use putty...

**Comment by vbjay** [comment]
Doesn't mean others don't.

**Comment by gerhardol** [comment]
Yes, this is just something to implement. When embedding PuTTY, it is easy to reference the binaries, no need to check for version etc..
The support needed does not have to be link to download PuTTY and a check at startup that if PuTTY is active and no longer present, go to settings (similar as for Git). (There could be a warning if PuTTY is older than recommended too).

Git was just more important to unbundle.

**Comment by gerhardol** [comment]
Suggestions:
 * Add link to PuTTY page, either a new page in the wiki (as for Git) or directly to https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html 
 * Remove PuTTY from installer (and repo)
 * Change default path in PuTTY settings, no longer related to GE path?
 * check PuTTY path when starting up, if PuTTy is selected, raise error (similar as for Git) if path does not exist

**Comment by vbjay** [comment]
I think we should have a link that sends them to the official installer.
We should show a settings check that says you have putty configured but
can't find any one of the paths.  I prefer putty.  All I did was point the
paths to my installed version.  Another idea is that we leverage
chocolatey.  Why can't we define a git extensions package that depends on
the putty package and git package.  We could have repair buttons that run
```choco update putty```.

**Comment by gerhardol** [comment]
> I think we should have a link that sends them to the official installer.

I suggest in settings, not installer for now
Maybe have a page with suggestions in the installer eventually?

**Comment by vbjay** [comment]
I meant the settings page in the program.  Yes I agree that we should get rid of the binaries.  I want the same pull request to not dump putty so my suggestion was to change the settings page to alert that putty path X can't be found.  Have it try to find the paths by looking for default install folder.  If that fails, pop a dialog asking the user for where the file is.

**Comment by RussKie** [comment]
@gitextensions/git-extensions-source do we want to tackle it for v3?

**Comment by gerhardol** [comment]
Ship v3 first

**Comment by vbjay** [comment]
Yep. With my chocolatey management system this will happen but it won't
hurt to leave it in for now.

**Comment by gerhardol** [comment]
> The installer should remember whether GE shall work with openSSH or with PuTTY.

Probably better removing PuTTY, then there is no option. 
This will mean that OpenSSH is default for new installations.

**Comment by vbjay** [comment]
I don't think you get the idea. They can pick putty or openssh if they pick
putty it makes sure chocolatey is installed and then runs Choco upgrade
putty -y.

**Comment by gerhardol** [comment]
> I don't think you get the idea

If you install GE and do not have PuTTY installed, OpenSSH will be the default. 
Maybe the settings can suggest that PuTTY is used if detected,regardless of installation method for PuTTY.

I knew this discussion would come up, otherwise I would have removed PuTTY already.
(PuTTY is my favorite terminal emulator, but I do not use it with GE.)

**Comment by vbjay** [comment]
I do. I also hate that we include it as a binary in our setup. The user
should not be beholden to us to make sure putty is up to date. With
chocolatey we are saying let chocolatey do it. So if a new version of putty
is released, they will get the latest version without waiting for us to
release a new version just so putty or other dependencies can be updated.
